RegisterPayments/handler.ts

```diff
@@ -0,0 +1,325 @@
+                  data: {
+                    ..._.data,
+                    singoloImportoPagato
+                  },
+                  elements: {
+                    ..._.elements,
+                    datiSingoloPagamento
+                  }
+                }))
+            )
+          )
+            .chain(_1 =>
+              fromOption(new Error("Missing dataEsitoSingoloPagamento"))(
+                fromNullable(
+                  _1.elements.datiSingoloPagamento
+                    .getElementsByTagName("dataEsitoSingoloPagamento")
+                    .item(0)
+                )
+                  .mapNullable(getElementTextContent)
+                  .map(dataEsitoSingoloPagamento => ({
+                    data: {
+                      ..._1.data,
+                      dataEsitoSingoloPagamento
+                    },
+                    elements: _1.elements
+                  }))
+              )
+            )
+            .chain(_1 =>
+              fromOption(new Error("Missing identificativoUnivocoRiscossione"))(
+                fromNullable(
+                  _1.elements.datiSingoloPagamento
+                    .getElementsByTagName("identificativoUnivocoRiscossione")
+                    .item(0)
+                )
+                  .mapNullable(getElementTextContent)
+                  .map(identificativoUnivocoRiscossione => ({
+                    data: {
+                      ..._1.data,
+                      identificativoUnivocoRiscossione
+                    },
+                    elements: _1.elements
+                  }))
+              )
+            )
+            .chain(_1 =>
+              fromOption(new Error("Missing causaleVersamento"))(
+                fromNullable(
+                  _1.elements.datiSingoloPagamento
+                    .getElementsByTagName("causaleVersamento")
+                    .item(0)
+                )
+                  .mapNullable(getElementTextContent)
+                  .map(causaleVersamento => ({
+                    data: { ..._1.data, causaleVersamento },
+                    elements: _1.elements
+                  }))
+              )
+            )
+            .chain(_1 =>
+              fromOption(new Error("Missing datiSpecificiRiscossione"))(
+                fromNullable(
+                  _1.elements.datiSingoloPagamento
+                    .getElementsByTagName("datiSpecificiRiscossione")
+                    .item(0)
+                )
+                  .mapNullable(getElementTextContent)
+                  .map(datiSpecificiRiscossione => ({
+                    data: {
+                      ..._1.data,
+                      datiSpecificiRiscossione
+                    },
+                    elements: _1.elements
+                  }))
+              )
+            )
+            .chain(_1 =>
+              fromOption(new Error("Missing commissioniApplicatePSP"))(
+                fromNullable(
```

Comment:
it looks like we have a repeating pattern, maybe it's worth to extract this as a method itself (fromOption / mapNullable / map)...

Reply:
Fixed in 67bc7ae